readthedocs_ext/_templates/readthedocs-insert.html.tmpl

```diff
@@ -18,16 +18,17 @@ http://docs.readthedocs.org/en/latest/canonical.html
 
 <link rel="stylesheet" href="{{ rtd_css_url }}" type="text/css" />
 
-<script type="text/javascript" src="{{ pathto('_static/readthedocs-data.js', 1) }}"></script>
+<script type="application/json" id="READTHEDOCS_DATA">{{ rtd_data | tojson }}</script>
```

Comment:
I'm a little worried about this possibly breaking things that users are depending on (eg. they are including this file themselves to get our data back out). I wonder if we should continue to build the file but not link it, at least for some time.

Reply:
While it is possible that this will break things for some users, I don't think it will do so in a systemic way for many users (say by breaking a theme). We maintain the global variable which was set in `readthedocs-data.js` for backwards compatibility and given that some page and suffix data was modified in that global below, nobody could correctly read the variable without parsing both the HTML and JS.

Reply:
Sure, but only the `page` was set in the HTML instead of the JS, so if they just were reading eg. the project or version, it would have worked fine. I think continuing to ship the file for now seems easy enough and minimal risk.

Reply:
I will see if it's possibly to produce the file from an extension rather than from a builder. When producing the file, perhaps I'll add a note into the file noting that it is deprecated and will be removed.

Reply:
Don't spend too much time if it's a pain, but shipping static files is pretty standard for extensions, so I think it should be simple enough.

Reply:
I just want to use the same function where the inline JSON is generated so they're the same.

Reply:
It's actually a little difficult to do this in a consistent way. I might skip it.

* I want to write that file, but only once per build
* I need the context which is in the `html-page-context` but I don't want to rewrite the same file for every page.
* I can't just detect if the file exists because the file might be there if the environment is re-used.